modules/apm/src/main/java/org/elasticsearch/telemetry/apm/APMMeterRegistry.java

```diff
@@ -170,8 +170,9 @@ public LongHistogram getLongHistogram(String name) {
 
     public void setProvider(Meter meter) {
         try (ReleasableLock lock = registerLock.acquire()) {
+            this.meter = meter;
```

Comment:
Is this change unrelated? I don't see setProvider actually called in the tests changed here. Separately, why is Meter passed into the ctor but also set here?

Reply:
> I don't see setProvider actually called in the tests changed here.

It's called when enabled/disabled.  That's what the tests should have been checking, but we were getting the same instance of `OpenTelemetry.noop()` so all the tests passed.

I noticed it when I fixed the boolean in `testMeterIsSetUponConstruction`.

> Separately, why is Meter passed into the ctor but also set here?

It's passed in the constructor so that if it metering is enabled, we start building up instruments with the correct meter provider, rather than building it up with the no-op provider and then changing it to the otel provider.